### ISSUE TITLE
Add setup.py script

### ISSUE THREAD

**Issue by MichaelClerx** [martinjrobins/pints#61]
We need two things:

1. Normal users should be able to 'install' pints, so that they can use `import pints` from anywhere in their project
2. Developers/users should be able to use the dev version (without getting confusion about two different libraries on the same system).

@jonc125 do you know of any 'best practices' for this?

**Comment by MichaelClerx** [comment]
The more general problem is: How do you treat Python dependencies on unfinished projects? 

So if we have two projects A and B, where both include tests, documentation, etc. and a directory (`A/a` and `B/b`) containing the Python `__init__.py` file. If A depends on B, you want some system where code in A can do `import b` and have access to `B/b`. But you don't want all the tests etc. for B to be duplicated in A

**Comment by jonc125** [comment]
What we tend to do is:
- Make the package installable with a `setup.py` file using [setuptools](http://setuptools.readthedocs.io/en/latest/setuptools.html)
- Set up a virtualenv (or conda env) for the overall project (`A` and `B` combined)
    - Or you could have one virtualenv for development just on `B`, and another for work on `A` which also has the dev version of `B` installed in it, in case testing `B` requires packages not needed for testing `A`
    - I tend to use anaconda these days and start off a project with `conda create -n proj python=3.6` or whatever
- Install your packages using `pip install -e .` for developers
    - Once this is done for `B`, code in `A` will be able to do `import B` and get the code as it exists in the working copy at that time - the 'installed' version is just a transparent reference to that
- Manage versions of dependencies explicitly using `pip-tools`, and install from requirements files with `pip install -r` - I'll be adding an example of this to the `cellmlmanip` repo shortly

**Comment by MichaelClerx** [comment]
Thanks!
I particularly like `pip install -e`, that seems to be a simple step that solves most of our problems :-)

**Comment by MichaelClerx** [comment]
https://stackoverflow.com/questions/30306099/pip-install-editable-vs-python-setup-py-develop
